fuzzers/065-gtp-common-pips/generate.tcl

```diff
@@ -0,0 +1,369 @@
+            set dst_wire [lindex $dst 1]
+
+            lappend todos $dst_wire
+        }
+
+        set todos_length [llength $todos]
+        if {$todos_length == 0} {
+            continue
+        }
+
+        puts "All todos for $tile_type / $wire"
+        foreach dst_wire $todos {
+            puts "  - $dst_wire"
+        }
+
+        set todos [shuffle_list $todos]
+
+        set target_node [get_nodes -of_objects [get_site_pins -filter {DIRECTION == IN} -of_objects $net]]
+        puts "Target node: $target_node"
+        route_design -unroute -nets $net
+
+        # Find an output in the todo list that can drive.
+        foreach dst_wire $todos {
+            if { [dict exists $used_destinations "$tile/$dst_wire"] } {
+                puts "Not routing to $tile / $dst_wire, in use."
+                continue
+            }
+
+            set origin_wire [get_wires "$tile/$dst_wire"]
+            set origin_node [get_nodes -of_objects $origin_wire]
+            if {[llength $origin_node] == 0} {
+                error "Failed to find node for $tile/$dst_wire."
+            }
+
+            set old_net [get_nets -of_objects $origin_node -quiet]
+            if {$old_net != {}} {
+                puts "Unrouting the old net: $old_net"
+                route_design -unroute -nets $old_net
+            }
+
+            # Route the net through the desired node
+            puts "Attempting to route to $dst_wire for net $net."
+            set route_status [route_via $net [list $origin_node] 0]
+
+            if {$route_status == 0} {
+                puts "WARNING: route failed, continue with next todo"
+                continue
+            }
+
+            puts "Target node: $target_node"
+            puts "Origin wire: $origin_wire, Old origin wire: $old_origin_wire"
+            puts "Origin node: $origin_node, Old origin node: $old_origin_node"
+
+            dict set used_destinations "$origin_wire" 1
+
+            break
+        }
+    }
+
+    set todo_map [load_todo "srcs"]
+    set pll_nets [get_nets_with_todo_pip_wires "srcs" "pll_clock" "HCLK_GTP_CK_IN" $used_destinations true true]
+    set mmcm_nets [get_nets_with_todo_pip_wires "srcs" "mmcm_clock" "HCLK_GTP_CK_IN" $used_destinations true true]
+    set cmt_nets [dict merge $pll_nets $mmcm_nets]
+    puts "CMT nets: $cmt_nets"
+    dict for {net tile_wire} $cmt_nets {
+        set tile [lindex $tile_wire 0]
+        set wire [lindex $tile_wire 1]
+        set dsts [dict keys $todo_map]
+        set tile_type [get_property TILE_TYPE [get_tiles $tile]]
+        set todos {}
+
+        set old_origin_dst_wire [get_wires -of_objects $net -filter {TILE_NAME =~ "*GTP_COMMON_MID*"}]
+        if {$old_origin_dst_wire == {}} {
+            continue
+        }
+
+        puts "Rerouting net $net at $tile / $wire (type $tile_type)"
+        puts "Previous target wire: $old_origin_dst_wire"
+
+        set old_origin_node [get_nodes -of_objects $old_origin_dst_wire]
```

Comment:
Comment around this stuff

Reply:
Done.
Removed the `old_origin_node` as it was actually unused.